### ISSUE TITLE
Build failed on macOS 10.13.3

### ISSUE THREAD

**Issue by 191919** [mpv-player/mpv#5604]
### mpv version and platform

mpv git HEAD on macOS 10.13.3 with Xcode 9.2:

`clang -v` output:

``
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin17.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
``

`xcode-select -p` output:

`/Applications/Xcode.app/Contents/Developer`

### Reproduction steps

./waf configure && ./waf

### Expected behavior

A successful build.

### Actual behavior

The first problem was about macOS_swift_bridge.h. `./waf` output:

``
[ 19/221] Processing osdep/macOS_swift.o osdep/macOS_swift.h osdep/macOS_swift.swiftmodule: osdep/macOS_mpv_helper.swift video/out/cocoa-cb/events_view.swift video/out/cocoa-cb/video_layer.swift video/out/cocoa-cb/window.swift video/out/cocoa_cb_common.swift -> build/osdep/macOS_swift.o build/osdep/macOS_swift.h build/osdep/macOS_swift.swiftmodule
macOS_swift_bridge.h
<unknown>:0: error: bridging header 'macOS_swift_bridge.h' does not exist
``

I worked around it by assigning the full path of `macOS_swift_bridge.h` to the variable `bridge` in the script file `wscript_build.py`.

The second problem:

``
[221/221] Linking build/mpv
ld: warning: directory not found for option '-LToolchains/XcodeDefault.xctoolchain/usr/lib/swift_static/macosx'
ld: library not found for -lswiftCoreData for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
``

I manually modified the library path to `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift_static/macosx` and the build passed.

**Comment by Akemi** [comment]
the `config.log` would be helpful and the output with `-v`. also xcode and swift version please.

**Comment by 191919** [comment]
swift version:

``
Apple Swift version 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2)
Target: x86_64-apple-macosx10.9
/Applications/Xcode.app/Contents/Developer/usr/bin/lldb "--repl=-enable-objc-interop -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -color-diagnostics"
Welcome to Apple Swift version 4.0.3 (swiftlang-900.0.74.1 clang-900.0.39.2). Type :help for assistance.
``

`./waf -v` output for the first problem:

```
[ 19/221] Processing osdep/macOS_swift.o osdep/macOS_swift.h osdep/macOS_swift.swiftmodule: osdep/macOS_mpv_helper.swift video/out/cocoa-cb/events_view.swift video/out/cocoa-cb/video_layer.swift video/out/cocoa-cb/window.swift video/out/cocoa_cb_common.swift -> build/osdep/macOS_swift.o build/osdep/macOS_swift.h build/osdep/macOS_swift.swiftmodule
09:20:00 runner '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -c -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk -enable-objc-interop -emit-objc-header -parse-as-library -target x86_64-apple-macosx10.10 -swift-version 3 -O -module-name macOS_swift -emit-module-path /Users/xx/box/mpv/build/osdep/macOS_swift.swiftmodule -import-objc-header macOS_swift_bridge.h -emit-objc-header-path /Users/xx/box/mpv/build/osdep/macOS_swift.h -o /Users/xx/box/mpv/build/osdep/macOS_swift.o /Users/xx/box/mpv/osdep/macOS_mpv_helper.swift /Users/xx/box/mpv/video/out/cocoa-cb/events_view.swift /Users/xx/box/mpv/video/out/cocoa-cb/video_layer.swift /Users/xx/box/mpv/video/out/cocoa-cb/window.swift /Users/xx/box/mpv/video/out/cocoa_cb_common.swift -I. -I..'
<unknown>:0: error: bridging header 'macOS_swift_bridge.h' does not exist

Waf: Leaving directory `/Users/xx/box/mpv/build'
Build failed
 -> task in 'osdep/macOS_swift.o osdep/macOS_swift.h osdep/macOS_swift.swiftmodule' failed (exit status 1): 
	{task 4386898576: osdep/macOS_swift.o osdep/macOS_swift.h osdep/macOS_swift.swiftmodule macOS_mpv_helper.swift,events_view.swift,video_layer.swift,window.swift,cocoa_cb_common.swift -> macOS_swift.o,macOS_swift.h,macOS_swift.swiftmodule}
''
```

As you could see, the full path of macOS_swift_bridge.h is not given to `swift`.

The following patch works:

```
--- a/wscript_build.py
+++ b/wscript_build.py
@@ -141,7 +141,7 @@ def build(ctx):
 
     def swift(task):
         src = ' '.join([x.abspath() for x in task.inputs])
-        bridge = ctx.path.find_node("osdep/macOS_swift_bridge.h")
+        bridge = ctx.path.find_node("osdep/macOS_swift_bridge.h").abspath()
         tgt = task.outputs[0].abspath()
         header = task.outputs[1].abspath()
         module = task.outputs[2].abspath()
```

The lines which are responsible for the second problems are:

```
Detected target OS:
os-darwin
----------------------------------------
Checking for macOS SDK
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.13.sdk
----------------------------------------
Checking for swift (Swift compiler)
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift
----------------------------------------
Checking for Swift Library
Toolchains/XcodeDefault.xctoolchain/usr/lib/swift_static/macosx
```

Again, the relative path was used, and the following patch works:

```
--- a/wscript_build.py
+++ b/wscript_build.py
@@ -141,7 +141,7 @@ def build(ctx):
 
     def swift(task):
         src = ' '.join([x.abspath() for x in task.inputs])
-        bridge = ctx.path.find_node("osdep/macOS_swift_bridge.h")
+        bridge = ctx.path.find_node("osdep/macOS_swift_bridge.h").abspath()
         tgt = task.outputs[0].abspath()
         header = task.outputs[1].abspath()
         module = task.outputs[2].abspath()
```

**Comment by Akemi** [comment]
i get that the relative path was used from your initial post. though i still wonder why in your case and why it works for me. hence why i asked for a full log.

**Comment by Akemi** [comment]
with a `-v` log from linking i could also see if [this](https://github.com/mpv-player/mpv/blob/master/wscript_build.py#L182) uses a relative path too in your case. this won't throw an error/warning when the file doesn't exist.

**Comment by Akemi** [comment]
i was able to reproduce this issue witht he help of the config log. it's a difference between waf 1.8.4 and newer versions. will be fixed via #5617.
